src/js/base/editing/Bullet.js

```diff
@@ -28,27 +28,18 @@ export default class Bullet {
     const paras = rng.nodes(dom.isPara, { includeAncestor: true });
     const clustereds = lists.clusterBy(paras, func.peq2('parentNode'));
 
-    $.each(clustereds, (idx, paras) => {
-      const head = lists.head(paras);
+    for (let i = 0; i < clustereds.length; i++) {
+      const head = lists.head(clustereds[i]);
       if (dom.isLi(head)) {
-        const previousList = this.findList(head.previousSibling);
-        if (previousList) {
-          paras
-            .map(para => previousList.appendChild(para));
-        } else {
-          this.wrapList(paras, head.parentNode.nodeName);
-          paras
-            .map((para) => para.parentNode)
-            .map((para) => this.appendToPrevious(para));
-        }
```

Comment:
It seems that some logic has changed rather than fixing the way of looping. Could you check that the tests pass locally?

Reply:
I have a project at work with a deadline on 16th may.
If everything goes as planned I might have time to look at it this weekend, otherwise I will look at it as soon as possible after the 16th.